### PR TITLE
feat: integrate iTX and tracing refactor

### PR DESCRIPTION
**Pull request by aqrln**
https://github.com/prisma/prisma-engines/pull/5001

### CODE REVIEW

**Comment by aqrln** [comment]
ci test all

File: packages/client/tests/functional/tracing/tests.ts [hunk 1]
```diff
@@ -635,7 +717,13 @@ testMatrix.setupTestSuite(
       test('should trace $disconnect', async () => {
         await _prisma.$disconnect()
 
-        await waitForSpanTree({ name: 'prisma:client:disconnect' })
+        await waitForSpanTree({
+          name: 'prisma:client:disconnect',
+          // There's not disconnect method in the binary engine, we terminate the process instead.
```
Comment (aqrln) [inline packages/client/tests/functional/tracing/tests.ts:722]:
```suggestion
          // There's no disconnect method in the binary engine, we terminate the process instead.
```

File: packages/client/tests/functional/extensions/itx.ts [hunk 2]
```diff
@@ -230,39 +229,50 @@ testMatrix.setupTestSuite(
       ).resolves.not.toThrow()
     })
 
-    test('middleware exclude from transaction also works with extended client', async () => {
-      const xprisma = prisma.$extends({})
+    // This test can lead to a deadlock on SQLite because we start a write transaction and a write query outside of it
+    // at the same time, and completing the transaction requires the query to finish. This leads a SQLITE_BUSY error
+    // after 5 seconds if the transaction grabs the lock first. For this test to work on SQLite, we need to expose
+    // SQLite transaction types in transaction options and make this transaction DEFERRED instead of IMMEDIATE.
```
Comment (jkomyno) [inline packages/client/tests/functional/extensions/itx.ts:235]: Do we have a Linear task for that?

Comment (aqrln) [reply packages/client/tests/functional/extensions/itx.ts:235]: no, it would be a full blown feature request, not a chore task

**Review by jkomyno — APPROVED** [review]

**Pull request merged**